### DISCUSSION TITLE
`input.conf`: help running one command strictly after another

### DISCUSSION THREAD

**Discussion by david-perez** [mpv-player/mpv#14688] [Q&A]
I've written a script that takes as input a file, prompts the user for a location using `dmenu`, and moves the file to the selected location. I'd like `mpv` to run it on the currently playing video and then start playing the next video in the playlist.

However, I can't figure out how to make `mpv` run a command, _and only after it exits_, run `playlist-next`.

I execute `mpv` as:

```sh
mpv --idle --force-window \
    --loop-file=inf \
    $(ls -1 --sort time --reverse *.mp4)
```

Let's take as an example the `sleep` command.

`input.conf`:

```
1 run "sleep 5" ; playlist-next
```

This _immediately_ plays the next video without waiting 5 seconds. [Docs say](https://mpv.io/manual/master/#command-interface-[-]]]):

> The program is run in a detached way. mpv doesn't wait until the command is completed, but continues playback right after spawning it.

> To get the old behavior, use `/bin/sh` and `-c` as the first two arguments.

But this doesn't work either:

```
1 run "/bin/sh" "-c" "sleep 5" ; playlist-next
```

Adding [the `sync` command prefix](https://mpv.io/manual/master/#command-interface-sync):

```
1 sync run "/bin/sh" "-c" "sleep 5" ; playlist-next
```

doesn't work either.

I also tried using [`subprocess`](https://mpv.io/manual/master/#command-interface-subprocess) instead of `run`, which supposedly "does not detach the process":

```
1 subprocess "sleep 5" ; playlist-next
```

but the behavior is identical.

**Comment by guidocella** [comment]
Get the old behavior refers to calling the shell, not to waiting until the command is completed. `subprocess` can't be used in input.conf, you need to call it from a Lua script.
